exasol_transformers_extension/udfs/models/filling_mask_udf.py

```diff
@@ -134,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
         :return: The dataframe where the model_df is formatted with the
         prediction results
         """
-        preds, labels = self._predict_model(model_df)
-        pred_df = self._prepare_prediction_dataframe(model_df, preds, labels)
+        pred_df_list = self._predict_model(model_df)
+        pred_df = self._prepare_prediction_dataframe(model_df, pred_df_list)
         return pred_df
 
-    def _predict_model(self, model_df: pd.DataFrame) -> \
-            Tuple[List[float], List[str]]:
+    def _predict_model(self, model_df: pd.DataFrame) -> List[pd.DataFrame]:
         """
         Predict the given text list using recently loaded models, return
         probability scores and filled texts
 
         :param model_df: The dataframe to be predicted
 
-        :return: A tuple containing prediction score list and filled texts list
+        :return: List of dataframe includes prediction details
         """
+        top_k = int(model_df['top_k'].iloc[0])
         text_data_raw = list(model_df['text_data'])
-        text_data_with_valid_mask_token = \
-            [text_data.replace(self.mask_token,
-                               self.last_created_pipeline.tokenizer.mask_token)
-             for text_data in text_data_raw]
-        results = self.last_created_pipeline(text_data_with_valid_mask_token)
+        text_data_with_valid_mask_token = [
+            text_data.replace(
+                self.mask_token,
+                self.last_created_pipeline.tokenizer.mask_token
+            ) for text_data in text_data_raw]
```

Comment:
extract to function

Reply:
These refactorings are actually part of the refactoring ticket #19  . All model udfs have these methods, somehow we collect them in a common class

Reply:
I think, this is specific to the filling mask task, so I think, it is a good idea to do it here